jigglypuffRL/common/values.py

```diff
@@ -34,7 +34,39 @@ def __init__(self, state_dim, action_dim=None, val_type="V", hidden=(32, 32)):
         self.model = _get_val_model(mlp, val_type, state_dim, hidden, action_dim)
 
 
-value_registry = {"mlp": MlpValue}
+def CNNValue(BaseValue):
+    """
+    CNN Value
+    :param state_dim: (int) state dimension of environment
+    :param action_dim: (int) action dimension of environment
+    :param history_length: (int) length of history of states
+    :param val_type: (str) type of value function.
+        'V' for V(s), 'Qs' for Q(s), 'Qsa' for Q(s,a)
+    :param hidden: (tuple or list) sizes of hidden layers
+    :param disc: (bool) discrete action space?
+    """
+    def __init__(
+        self, state_dim, action_dim, history_length=4, val_type="Qs",
+        hidden=(256, 256), disc=True
+    ):
+        super(CNNValue, self).__init__(disc, state_dim, action_dim, hidden)
+
+        self.action_dim = action_dim
+
+        self.conv, output_size = cnn((history_length, 16, 32))
+
+        self.fc = _get_val_model(
+            mlp, val_type, output_size, hidden, action_dim
+        )
```

Comment:
There will only be one linear hidden layer for CNN policies/values

Reply:
Wait, we need two hidden layers: one from convolutional output to 256 hidden units and the other from 256 hidden units to the final output. So the hidden parameter should be (256) instead of (256, 256), is that right?

Reply:
Yes, there will be two layers but one hidden and one output.